test/requestContextPlugin.e2e.spec.js

```diff
@@ -0,0 +1,106 @@
+const fastify = require('fastify')
+const request = require('superagent')
+const { fastifyRequestContextPlugin } = require('../lib/requestContextPlugin')
+const { TestService } = require('./internal/testService')
+
+function initAppPostWithPrevalidation(endpoint) {
+  return new Promise((resolve) => {
+    const app = fastify({ logger: true })
+    app.register(fastifyRequestContextPlugin, { hook: 'preValidation' })
+
+    const preValidationFn = (req, reply, done) => {
+      const requestId = Number.parseInt(req.body.requestId)
+      req.requestContext.set('testKey', `testValue${requestId}`)
+      done()
+    }
+
+    app.route({
+      url: '/',
+      method: ['GET', 'POST'],
+      preValidation: preValidationFn,
+      handler: endpoint,
+    })
+
+    app.listen(8085, '0.0.0.0', (err, address) => {
```

Comment:
please set 0 as the port and use `app.server.address()` to get the actual port.
There is no need to use `0.0.0.0` either.

Reply:
Done.